### PR TITLE
Add a single file based cache module

### PR DESCRIPTION
**Pull request by lambdalisue**
While `System.Cache.File` create individual files for individual keys, I made a new cache module which use a single file to store a dictionary of `System.Cache.Memory`.

Users can use this new cache module as a permanent memory cache module.

### CODE REVIEW

**Comment by ujihisa** [comment]
s/System.File.Cache/System.Cache.File/ > d75b4b9

**Comment by lambdalisue** [comment]
Thanks ;-)

File: doc/vital-system-cache-file.txt [hunk 1]
```diff
@@ -16,6 +16,11 @@ Methods			|Vital.System.Cache.File-methods|
 INTRODUCTIONS				*Vital.System.Cache.File-intro*
 
 *Vital.System.Cache.File* is a file based unified cache system.
+It store a value into a file in a cache directory, mean that individual files
```
Comment (ujihisa) [inline doc/vital-system-cache-file.txt]: s/store/stores/

File: doc/vital-system-cache-file.txt [hunk 2]
```diff
@@ -52,10 +57,22 @@ FUNCTIONS				*Vital.System.Cache.File-functions*
...
 
 	Create a new instance of System.Cache.File.
-	It require 'cache_dir' option in {options}. If no 'cache_dir' is
+	It requires 'cache_dir' option in {options}. If no 'cache_dir' is
 	specified, it will throw an exception.
 	The 'cache_dir' will automatically be created if missing.
 
+load({filename}[, {default}])		*Vital.System.Cache.File.load()*
+
+	Load a vim object from a {filename}. It use |sandbox| and |eval| internally
```
Comment (ujihisa) [inline doc/vital-system-cache-file.txt]: s/use/uses/

File: doc/vital-system-cache-singlefile.txt [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+Usage			|Vital.System.Cache.SingleFile-usage|
+Functions		|Vital.System.Cache.SingleFile-functions|
+Methods			|Vital.System.Cache.SingleFile-methods|
+
+
+==============================================================================
+INTRODUCTIONS				*Vital.System.Cache.SingleFile-intro*
+
+*Vital.System.Cache.SingleFile* is a single file based cache system.
+It store a cache dictionary into a single cache file, mean that it is similar
```
Comment (ujihisa) [inline doc/vital-system-cache-singlefile.txt]: s/store/stores/

File: doc/vital-system-cache-singlefile.txt [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+to |Vital.System.Cache.Memory| but the cache is permanent.
```
Comment (ujihisa) [inline doc/vital-system-cache-singlefile.txt]: s/permanent/persistent/ sounds more idiomatic.

File: doc/vital-system-cache-singlefile.txt [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+	missing.
+
+
+==============================================================================
+METHODS					*Vital.System.Cache.SingleFile-methods*
+
+dump()				*Vital.System.Cache.SingleFile.dump()*
+
+	Dump a cache content into the 'cache_file'.
+	It will automatically be called when the cache content is changed.
```
Comment (ujihisa) [inline doc/vital-system-cache-singlefile.txt]: `It will be automatically called` sounds more natural.

**Comment by lambdalisue** [comment]
@ujihisa Thanks a lot. I'll follow the points when you said "Done" ;-)

File: test/System/Cache/SingleFile.vimspec [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+      let cache_file = s:P.join(cache_dir, 'cachefile')
+      call s:C.new({ 'cache_file': cache_file })
+      Assert True(isdirectory(cache_dir))
+      if isdirectory(cache_dir)
+        call s:F.rmdir(cache_dir, 'r')
+      endif
+    End
+
+    It should raise an exception if 'cache_file' option is not specified.
+      Throw call s:C.new(), '"cache_file" option is empty'
```
Comment (thinca) [inline test/System/Cache/SingleFile.vimspec]: これたぶんちゃんとテストできてないですね。書き方が間違っているので、全然違う例外が出ていると思います。

Comment (lambdalisue) [inline test/System/Cache/SingleFile.vimspec]: Fixed.

**Comment by ujihisa** [comment]
:+1: :sushi:

**Comment by ujihisa** [comment]
:clap:

**Comment by lambdalisue** [comment]
Thanks!

**Pull request merged**